### PR TITLE
NavBar login user information

### PR DESCRIPTION
**Pull request by HongbinZheng**
User info now won't lost after refresh page. And all user info will store in this.props.auth.user

### CODE REVIEW

File: front_end/src/actions/authActions.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const loginUser = userData => dispatch => {
     .post("/api/users/login", userData)
     .then(res => {
       // save to localStorage
-      const { token, email } = res.data;
+      const { token, email,firstname,lastname,rewardPoints } = res.data;
```
Comment (SandroSal) [inline front_end/src/actions/authActions.js]: I think we retrieve rewardPoints thru profile.profile_info

File: front_end/src/actions/authActions.js [hunk 2]
```diff
@@ -38,7 +38,10 @@ export const loginUser = userData => dispatch => {
         id: decoded.id,
         iat: decoded.iat,
         exp: decoded.exp,
-        email: email
+        email: email,
+        firstname: firstname,
+        lastname: lastname,
+        rewardPoints: rewardPoints
```
Comment (SandroSal) [inline front_end/src/actions/authActions.js]: same comment as above

File: front_end/src/actions/authActions.js [hunk 3]
```diff
@@ -52,6 +55,21 @@ export const loginUser = userData => dispatch => {
...
+export const getCurrentUser = decode => dispatch => {
+  axios.get("api/users/current").then(res =>{
+    const decoded = {
+      id: decode.id,
+      iat: decode.iat,
+      exp:decode.exp,
+      firstname:res.data.firstName,
+      lastname:res.data.lastName,
+      email:res.data.email,
+      rewardPoints:res.data.rewardPoints
```
Comment (SandroSal) [inline front_end/src/actions/authActions.js]: and again

File: front_end/src/components/layout/Navbar.js [hunk 4]
```diff
@@ -2,10 +2,12 @@ import React, { Component } from "react";
 import { Link } from "react-router-dom";
 import PropTypes from "prop-types";
 import { connect } from "react-redux";
-import { logoutUser } from "../../actions/authActions";
+import { logoutUser,getCurrentUser } from "../../actions/authActions";
 import { clearCurrentProfile } from "../../actions/profileActions";
 import AnchorLink from "react-anchor-link-smooth-scroll";
 import NavbarMenu from "./NavbarMenu";
+import axios from "axios";
+import jwt_decode from "jwt-decode";
```
Comment (SandroSal) [inline front_end/src/components/layout/Navbar.js]: is jwt_decode an alternative to redux store?

Comment (DaleCS) [reply front_end/src/components/layout/Navbar.js]: iirc, it's used to distributed authentication tokens to logged in users.

Comment (SandroSal) [reply front_end/src/components/layout/Navbar.js]: but in this case, does it include only the token or all the other info (name, email, points)?

File: front_end/src/components/layout/Navbar.js [hunk 5]
```diff
@@ -24,7 +25,13 @@ class Navbar extends Component {
     this.props.logoutUser();
   }
 
-  render() {
+  componentWillMount(){
```
Comment (SandroSal) [inline front_end/src/components/layout/Navbar.js]: do we still need this, if we do it through the auth action?

Comment (HongbinZheng) [reply front_end/src/components/layout/Navbar.js]: yes, cause when we refresh it, it won't call the auth action, we need to call the profile action to reload the `setCurrentUser()` then we can keep the user info. And we need the profile action to get the current profile to pass to `setCurrentUser()`

**Pull request merged**